types/types.go

```diff
@@ -671,6 +671,9 @@ type SystemContext struct {
 	CompressionFormat *compression.Algorithm
 	// CompressionLevel specifies what compression level is used
 	CompressionLevel *int
+	// Give priority to pulling zstd images if multiple images are present unless explicitly
+	// configured false.
+	GivePriorityToZstdImages OptionalBool
```

Comment:
how would it work if we ever add another compression level?

Isn't it better to have something like `PreferGzip` to maintain the current behavior?

I think the new zstd priority code should be the default, and have a way to opt-out.  In the long term we want everyone to move to using zstd so IMO it makes more sense to make it the default and have a possibility to opt-out when needed

Reply:
@giuseppe `PreferGzip` sounds munch better but as discussed above, not using this `OptionalBool` anymore , I'll add a new option to `copy.Options` and will pass it down to the `internal/manifest.List`'s ChooseInstance method so will add a `PreferGzip` there.

Reply:
I agree.

Reply:
I think, at this point, we are somewhat experimenting and the options are therefore very specific. `PreferZstdOverGzip` is not a good long-term API, but it’s something we can do now without too much design.

Alternatively, we _could_ draft what this should look like in some end state, generalized to multiple compression formats, and the like:
```go
    EnsureCompressionVariantsExist Set<CompressionVariant> // add these if they don’t exist
    StripCompressionVariants Set<CompressionVariant> // remove these if they exist
    StripToOneCompressionVariant bool // if there are multiple variants, only preserve one
    CompressionVariantPreferenceOrder []CompressionVariant // relevant for StripToOneCompressionVariant and for CopySystemImage
```

---

But right now, I’m not too tempted to overthink this. Note that the `GivePriorityToZstdImages` / `PreferGzip` is currently _primarily_ desired by @vrothberg just as a feature flag, to allow us to revert to the previous behavior, not because we expect users to _long_ term make these choices at all. Such a feature flag is inherently short-term and not worth _too_ much design, apart from the fact that we will need to keep it around forever due to API stability.

OTOH it’s not _quite_ short-term: we do need to give users some kind of choice for `CopySystemImage` to destinations other than local c/storage … or, um, maybe to recommend against using `CopySystemImage` in such situations completely.